src/coreclr/tools/aot/ILCompiler/ILCompiler_publish.csproj

```diff
@@ -3,7 +3,11 @@
   <PropertyGroup>
     <_IsPublishing>true</_IsPublishing>
     <RuntimeIdentifier>$(OutputRID)</RuntimeIdentifier>
-    <RuntimeIdentifier Condition="'$(BaseOS)' != ''">$(BaseOS)</RuntimeIdentifier>
+    <!--
+      If the output RID isn't the current SDK RID and we have a "base" RID, then the output RID isn't known to the SDK.
+      In that case, we need to set the RuntimeIdentifier to the base RID so the SDK can find a runtime pack for publishing.
+    -->
+    <RuntimeIdentifier Condition="'$(OutputRID)' != '$(NETCoreSdkRuntimeIdentifier)' and '$(BaseOS)' != '' ">$(BaseOS)</RuntimeIdentifier>
```

Comment:
Isn't BaseOS just the OS name without architecture part?

Reply:
Perhaps we want:
```suggestion
    <RuntimeIdentifier Condition="'$(OutputRID)' != '$(NETCoreSdkRuntimeIdentifier)' and '$(BaseOS)' != ''">$(BaseOS)-$(TargetArchitecture)</RuntimeIdentifier>
```

Reply:
`BaseOS` is the RID that the non-portable RID for source-build is based on. It's a full RID (it's very poorly named, I'm planning on renaming it after we enable VMR backflow and I can make the change in one place).

Reply:
Ah, makes sense. 👍 for renaming it something with RID suffix.